### PR TITLE
Added bindings for NvJpeg Single Image Decode.

### PR DESCRIPTION
**Pull request by MoFtZ**
Related to #422.

Initial bindings for NvJpeg with simple wrappers handles that need to be disposed.

### CODE REVIEW

File: Src/ILGPU.Algorithms/Runtime/Cuda/NvJpegLibrary.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        /// Creates a new NvJpeg state instance.
+        /// </summary>
+        public NvJpegState CreateState()
+        {
+            NvJpegException.ThrowIfFailed(
+                API.JpegStateCreate(LibHandle, out IntPtr stateHandle));
+            return new NvJpegState(API, stateHandle);
+        }
+
+        /// <inheritdoc cref="NvJpegAPI.GetImageInfo(IntPtr, ReadOnlySpan{byte}, out int, out NvJpegChromaSubsampling, out int[], out int[])"/>
```
Comment (m4rs-mt) [inline Src/ILGPU.Algorithms/Runtime/Cuda/NvJpegLibrary.cs:67]: Question: should we break these comments to ensure that the max-line length will not be exceeded?

File: Samples/AlgorithmsNvJpeg/Program.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
+﻿using ILGPU;
```
Comment (m4rs-mt) [inline Samples/AlgorithmsNvJpeg/Program.cs:1]: Question: is the copyright header missing?

**Review by m4rs-mt — CHANGES_REQUESTED** [review]
Great work! The only thing I'm afraid of right now is the sample image. I bought a publication license for the website which is not suitable for further publications. Can we replace this image with another one using a Creative Commons license?

File: Src/ILGPU.Algorithms/Runtime/Cuda/NvJpegLibrary.cs (hunk 1 above)
Comment (MoFtZ) [reply Src/ILGPU.Algorithms/Runtime/Cuda/NvJpegLibrary.cs:67]: Done - didn't realise I could wrap the cref.

File: Samples/AlgorithmsNvJpeg/Program.cs (hunk 2 above)
Comment (MoFtZ) [reply Samples/AlgorithmsNvJpeg/Program.cs:1]: Added.

**Comment by MoFtZ** [comment]
Fixed review comments.
Changed sample image to ILGPU logo 😄

**Comment by m4rs-mt** [comment]
Amazing work! Thanks a lot @MoFtZ 🏅

**Review by m4rs-mt — APPROVED** [review]

**Pull request merged**